p74/src/bin/rustcrypto_sha2.rs

```diff
@@ -0,0 +1,44 @@
+use sha2::{Digest, Sha256, Sha512};
+use std::env;
+use std::fs::File;
+use std::io::{self, Read};
+
+fn main() -> io::Result<()> {
+    // Collect the command line arguments
+    let args: Vec<String> = env::args().collect();
+    if args.len() != 3 {
+        eprintln!("Usage: {} <algorithm> <file_path>", args[0]);
+        std::process::exit(1);
+    }
+
+    let algorithm = &args[1];
+    let file_path = &args[2];
+
+    // Open and read the file
+    let mut file = File::open(file_path)?;
+    let mut buffer = Vec::new();
+    file.read_to_end(&mut buffer)?;
```

Comment:
Ideally, hashing should be done in chunks without reading the whole file into memory.